testHelper/simulation.go

```diff
@@ -129,105 +130,175 @@ func StartSim(GivenNodes string, UserAddedOptions map[string]string) *state.Stat
+	}
+
+	fmt.Println("SetupSim(", givenNodes, ",", userAddedOptions, ",", height, ",", electionsCnt, ",", roundsCnt, ")")
+
+	if userAddedOptions["--factomhome"] == "" {
+		// default to create a new home dir for each sim test if not specified
+		homeDir := GetSimTestHome(t)
+		err := os.MkdirAll(filepath.Join(homeDir, "/.factom/m2"), 0755)
+		if err != nil {
+			t.Fatal(err)
+		}
+		userAddedOptions["--factomhome"] = homeDir
+	}
+
+	ExpectedHeight = height
+	nodeLen := setNodeCounts(givenNodes)
+	state0 := StartSim(nodeLen, userAddedOptions)
 	state0.MessageTally = true
 
-	fmt.Printf("Starting timeout timer:  Expected test to take %s or %d blocks\n", calctime.String(), height)
+	{ // calculate & set test timeout
+		blkt := globals.Params.BlkTime
+		roundt := elections.RoundTimeout
+		et := elections.FaultTimeout
+		setTestTimeouts(state0, time.Duration(float64(((height+3)*blkt)+(electionsCnt*et)+(roundsCnt*roundt))*1.1)*time.Second)
+	}
+
 	StatusEveryMinute(state0)
-	WaitMinutes(state0, 1) // wait till initial DBState message for the genesis block is processed
-	creatingNodes(GivenNodes, state0, t)
 
-	l := len(GivenNodes)
-	t.Logf("Allocated %d nodes", l)
-	if len(engine.GetFnodes()) != l {
-		t.Fatalf("Should have allocated %d nodes", l)
-		t.Fail()
-	}
-	if Audits == 0 && Leaders == 0 {
-		// if no requested promotions then assume we loaded from a database and the test will check
+	if isDefaultSim(givenNodes) || state0.GetDBHeightAtBoot() != 0 {
+		t.Logf("Skip Node Promotion", nodeLen)
 	} else {
+		WaitMinutes(state0, 1) // wait till initial DBState message for the genesis block is processed
+		createAuthoritySet(givenNodes, state0, t)
+
+		if len(engine.GetFnodes()) != nodeLen {
+			t.Fail()
+		}
+
+		// swap identity if Fnode0 Should be a follower
+		if []rune(givenNodes)[0] == 'F' {
+			RunCmd(fmt.Sprintf("%d", 0))
+			RunCmd("z")
+			//WaitMinutes(state0, 2)
+			WaitBlocks(state0, 1)
+			RunCmd(fmt.Sprintf("%d", 0))
+			RunCmd(fmt.Sprintf("t%d", len(givenNodes)+1)) // attach the last generated Identity
+		}
+		// REVIEW: should we swap node0 identity & promote if configured for 'L' ?
+
 		CheckAuthoritySet(t)
 	}
+
+	if len(engine.GetFnodes()) != nodeLen {
+		t.Fatalf("Should have allocated %d nodes", nodeLen)
+	} else {
+		t.Logf("Allocated %d nodes", nodeLen)
+	}
+
 	return state0
 }
 
-func creatingNodes(creatingNodes string, state0 *state.State, t *testing.T) {
-	RunCmd(fmt.Sprintf("g%d", len(creatingNodes)))
-	WaitBlocks(state0, 3) // Wait for 2 blocks because ID scans is for block N-1
-	WaitMinutes(state0, 1)
-	// Wait till all the entries from the g command are processed
-	simFnodes := engine.GetFnodes()
-	nodes := len(simFnodes)
-	if len(creatingNodes) > nodes {
-		t.Fatalf("Should have allocated %d nodes", len(creatingNodes))
-	}
-	WaitForMinute(state0, 1)
+func promoteNodes(creatingNodes string) int {
+
 	for i, c := range []byte(creatingNodes) {
 		fmt.Println("it:", i, c)
 		switch c {
-		case 'L', 'l':
```

Comment:
I don't think we use lower anywhere - moving to try to standardize on caps

Reply:
What's the benefit to being case sensitive?